docs/editor.rst

```diff
@@ -1,4 +1,112 @@
 .. _caster-editor:
 
-Caster editor
+Caster Editor
 =============
+
+.. seealso::
+
+    This documents covers the editing of a *story graph*.
+    For the concepts of such a graph see :ref:`story-graph`.
+
+
+.. _caster-editor-script-cells:
+
+Script cells
+------------
+
+Markdown
+^^^^^^^^
+
+Markdown is a way to write formatted text in an easy but extensible manner.
+The written text will be converted to audio via :class:`~stream.models.TextToSpeech`.
+
+It is possible to make this text dynamic by using the value of a :class:`~stream.models.StreamVariable` within the text.
+
+.. code-block:: markdown
+
+    Hello ${var}`name`.
+    I hope you are doing fine.
+
+All possible macros are documented in :class:`~story_graph.markdown_parser.GencasterRenderer`.
+
+Audio
+^^^^^
+
+An audio cell allows to playback a :class:`~stream.models.AudioFile` on the stream in two ways
+
+- ``async`` will playback in the background and the story graph will continue with the execution
+- ``sync`` will pause the execution of the story graph until the file has been played back fully.
+
+
+Python
+^^^^^^
+
+Python is an universal scripting language and allows to
+
+- interact with the graph (e.g. set next node)
+- create or read out :class:`~stream.models.StreamVariable`
+- trigger dialogs on the frontend
+
+The execution is async which allows to wait for something to happen, e.g. for a stream variable through an user interaction with the help of :meth:`~story_graph.engine.Engine.wait_for_stream_variable`.
+
+.. code-block:: python
+
+    await wait_for_stream_variable('start')
+
+A stream variable can be accessed through a dictionary
+
+.. code-block:: python
+
+    # will return None if not set
+    name = vars.get('name')
+
+In order to set a stream variable, simply set the value in the ``vars`` dictionary
+
+.. code-block:: python
+
+    vars['day'] = 8 < datetime.now().hour < 20
+
+To trigger a dialog in the frontend use the ``yield`` command with a :class:`~stream.frontend_types.Dialog` instance.
+The exact arguments are documented in :ref:`frontend-types`.
+
+.. code-block:: python
+
+    yield Dialog(
+        title="Headline",
+        content=[
+            Text(text="Can we ask for your name?"),
+            Input(label="Name", key="name"),
+        ],
+        buttons=[Button.cancel(), Button.ok()],
+    )
+    # wait for user input
+    await wait_for_stream_variable("name")
+
+This will save the value that was inserted by the user into tho variable ``name``.
```

Comment:
This is so great. We can also have custom templates, opening dialogs with buttons like:
"change name" or so. Just a thought I wanted to share :)